src/components/Button/index.tsx

```diff
@@ -115,6 +115,9 @@ type ButtonProps = Partial<ChildrenProps> & {
 
     /** Boolean whether to display the right icon */
     shouldShowRightIcon?: boolean;
+
+    /** The custom icon to display to the right of the text */
+    customRightIcon?: React.ReactNode;
```

Comment:
I'm not sure that I'm following this. We already have an `iconRight` prop, can't we just reuse that and `shouldShowRightIcon`?

Reply:
I'm still not sure that I follow. I don't think we should add another Icon prop just to use different styles. Can we just compare the sizes instead?

```
large={buttonSize === CONST.DROPDOWN_BUTTON_SIZE.LARGE}
medium={buttonSize === CONST.DROPDOWN_BUTTON_SIZE.MEDIUM}
small={buttonSize === CONST.DROPDOWN_BUTTON_SIZE.SMALL}
```

Reply:
@luacmartins The current style of the button and the style of the icon are not consistent: 
- when the button has a size of large, the icon only has a size of medium
- when the button has a size other than large, it has a small size

So we can't just compare the sizes.

Reply:
So are you saying that when we have a split button, the logic should be:

```
large={isButtonSizeLarge}
medium={!isButtonSizeLarge}
```

but for non-split buttons it's supposed to be:

```
large={isButtonSizeLarge}
medium={!isButtonSizeLarge}
```

Is that right? If so, I think we can derive those from a combination of isSplitButton and the button size.

Reply:
@luacmartins i just updated code, please check again 🙏